packages/toolkit/src/lib/react-query-service/model/useInfiniteModelVersions.ts

```diff
@@ -0,0 +1,70 @@
+import {
+  InfiniteData,
+  UseInfiniteQueryResult,
+  useInfiniteQuery,
+} from "@tanstack/react-query";
+import type { Nullable } from "../../type";
+import {
+  ListModelVersionsResponse,
+  listModelVersionsQuery,
+} from "../../vdp-sdk";
+
+export function useInfiniteModelVersions({
+  accessToken,
+  enabledQuery,
+  modelName,
+  pageSize,
+  retry,
+}: {
+  accessToken: Nullable<string>;
+  enabledQuery: boolean;
+  modelName: string;
```

Comment:
@orangecoloured Most of time, to ensure we have more flexibility on the outer space, I will put this modelName as Nullable<string> but guard it below like

```
if (modelName && enabledQuery) {
    enabled = true;
  }
```